Libraries/LibJS/Parser.cpp

```diff
@@ -68,8 +68,11 @@ NonnullOwnPtr<Statement> Parser::parse_statement()
     case TokenType::Let:
     case TokenType::Const:
         return parse_variable_declaration();
-    case TokenType::For:
-        return parse_for_statement();
+    case TokenType::For: {
+        auto for_block = make<BlockStatement>();
+        for_block->append(parse_for_statement());
+        return for_block;
```

Comment:
Hmm, I don't see other parsers wrapping `ForStatement` in a `BlockStatement`.

Reply:
Oops, forgot to think of this from  the perspective of the conformance of the AST we produce. It's correct that this will result in somewhat unconventional AST and is quite hacky, I first thought about implementing some sort of queue for variables that need to enter the next scope and not the current one, but then realized that the for statement (where init, predicate and update are) is in another scope than the for loop's body, which renders this cumbersome.

If this is deemed too hacky, I'll just close the PR, I think we can live without `let` and `const` for for loops for a while.

Reply:
Or maybe we could still parse it normally and produce a normal AST but execute the loop as if it were in its own scope in `ForStatement::execute`?  That would render the AST disingenuous though.

Reply:
https://www.ecma-international.org/ecma-262/10.0/index.html#sec-for-statement-runtime-semantics-labelledevaluation this is where the correct behaviour is specified. Basically the interpreter should create a new lexical scope when executing the statement if the first element is a LexicalDeclaration.